src/Symfony/Component/Console/EventListener/ExceptionListener.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Symfony\Component\Console\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleExceptionEvent;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Console exception listener.
+ *
+ * Attempts to log exceptions or abnormal terminations of console commands.
+ *
+ * @author James Halsall <james.t.halsall@googlemail.com>
+ */
+class ExceptionListener implements EventSubscriberInterface
+{
+    protected $logger;
+
+    /**
+     * Constructor.
+     *
+     * @param LoggerInterface $logger A logger
+     */
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Handles console command exception.
+     *
+     * @param ConsoleExceptionEvent $event Console event
+     */
+    public function onKernelException(ConsoleExceptionEvent $event)
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $exception = $event->getException();
+
+        $this->logger->error($exception->getMessage(), array('exception' => $exception));
+    }
+
+    /**
+     * Handles termination of console command.
+     *
+     * @param ConsoleTerminateEvent $event Console event
+     */
+    public function onKernelTerminate(ConsoleTerminateEvent $event)
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $exitCode = $event->getExitCode();
+
+        if ($exitCode === 0) {
+            return;
+        }
+
+        $message = sprintf(
+            'Command `%s` exited with status code %d',
+            implode(' ', $_SERVER['argv']),
```

Comment:
Maybe it would offer value to others to add something to `Symfony\Component\Console\Input\Input` that basically generates the result of `implode(' ', $_SERVER['argv'])` but via the Input itself so it doesn't rely on globals?